client/components/account-card.jsx

```diff
@@ -139,6 +142,12 @@ export default class AccountCard extends React.Component {
     </h1>;
   }
 
+  handleUsernameClick(event) {
+    if (this.state.route.path === 'view-other-accounts') {
+      window.location.hash = `#user-account?username=${event.target.textContent}`;
+    }
+  }
```

Comment:
It should be possible to use the same route for the current user's account and the other user accounts. If you go to `github.com/jihengnie`, it's your own account, but i can also go to that same page to see your account.